### PR TITLE
chore(rust): enable `sentry-tracing` integration

### PR DESCRIPTION
**Pull request by thomaseizinger**
Using the `sentry-tracing` integration, we can automatically capture events based on what we log via `tracing`. The mapping is defined as follows:

- ERROR: Gets captured as a fatal error
- WARN: Gets captured as a message
- INFO: Gets captured as a breadcrumb
- `_`: Does not get captured at all

If telemetry isn't active / configured, this integration does nothing. It is therefore safe to just always enable it.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Setting as draft for now because I still need to test this.

**Comment by thomaseizinger** [comment]
> Setting as draft for now because I still need to test this.

Looking at the latest issues in sentry, this is already working pretty well!

File: rust/logging/src/sentry.rs
Comment (ReactorScram) [inline rust/logging/src/sentry.rs:1]: Why is this function its own module?

File: rust/telemetry/src/lib.rs [hunk 1]
```diff
@@ -66,7 +66,7 @@ impl Telemetry {
                 environment: Some(environment.into()),
                 // We can't get the release number ourselves because we don't know if we're embedded in a GUI Client or a Headless Client.
                 release: Some(release.into()),
-                traces_sample_rate: 1.0,
+                traces_sample_rate: 0.1,
```
Comment (ReactorScram) [inline rust/telemetry/src/lib.rs:70]: We'll still get all errors and messages, this is for just (lowercase 't') tracing, right?

**Review by ReactorScram — APPROVED** [review]
Thanks for getting this!

File: rust/telemetry/src/lib.rs (hunk 1 above)
Comment (thomaseizinger) [reply rust/telemetry/src/lib.rs:70]: Yeah I think this is for sending the "transactions" as Sentry calls them. The docs recommend to set this to < 1.0. Maybe 10% is even too much? Perhaps 1% is better?

File: rust/logging/src/sentry.rs
Comment (thomaseizinger) [reply rust/logging/src/sentry.rs:1]: Just figured separation of concerns but we can stick it into lib.rs!

**Comment by thomaseizinger** [comment]
@ReactorScram I think the "worst" thing that can happen is that we receive way too many events. I guess it is okay to just merge this and evaluate / tweak later?

File: rust/telemetry/src/lib.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/telemetry/src/lib.rs:70]: 10% is probably fine, I only trigger transactions for lifecycle events like starting connlib, so it won't happen often per system

Comment (thomaseizinger) [reply rust/telemetry/src/lib.rs:70]: I'll have to double check but I think tracing spans get picked up too. We don't have any on WARN or above though so it should be fine.

Comment (thomaseizinger) [reply rust/telemetry/src/lib.rs:70]: So yeah, all tracing spans of level INFO and above are captured as transactions.

**Pull request merged**